internal/nostr/nostr.go

```diff
@@ -614,53 +662,112 @@ func (svc *Service) startSubscription(ctx context.Context, subscription *Subscri
 			continue
 		}
 
-		for {
-			sub, err := relay.Subscribe(ctx, []nostr.Filter{filter})
-			if err != nil {
-				// TODO: notify user about subscription failure
-				svc.Logger.WithError(err).WithFields(logrus.Fields{
-					"subscriptionId": subscription.ID,
-					"relayUrl":       subscription.RelayUrl,
-				}).Error("Failed to subscribe to relay, retrying in 5s...")
-				time.Sleep(5 * time.Second) // sleep for 5 seconds
-				break
-			}
+		sub, err := relay.Subscribe(ctx, []nostr.Filter{*filter})
+		if err != nil {
+			// TODO: notify user about subscription failure
+			svc.Logger.WithError(err).WithFields(logrus.Fields{
+				"subscriptionId": subscription.ID,
+				"relayUrl":       subscription.RelayUrl,
+			}).Error("Failed to subscribe to relay, retrying in 5s...")
+			time.Sleep(5 * time.Second) // sleep for 5 seconds
+			continue
+		}
 
-			svc.subscriptionsMutex.Lock()
-			svc.subscriptions[subscription.ID] = sub
-			svc.subscriptionsMutex.Unlock()
+		svc.subscriptionsMutex.Lock()
+		svc.subscriptions[subscription.ID] = sub
+		svc.subscriptionsMutex.Unlock()
 
-			svc.Logger.WithFields(logrus.Fields{
-				"subscriptionId": subscription.ID,
-			}).Info("Started subscription")
+		svc.Logger.WithFields(logrus.Fields{
+			"subscriptionId": subscription.ID,
+		}).Info("Started subscription")
 
-			err = svc.processEvents(ctx, subscription, sub)
-			// closing relay as we reach here due to either
-			// halting subscription or relay error
-			if isCustomRelay {
-				relay.Close()
-			}
+		err = svc.processEvents(ctx, subscription, sub, onReceiveEOS, persistEvent)
+		// closing relay as we reach here due to either
+		// halting subscription or relay error
+		if isCustomRelay {
```

Comment:
this will not be hit if one of the `continue` statements get hit further above in this loop

Reply:
Fixed